### PR TITLE
Call avoidance

### PR DESCRIPTION
**Pull request by bjkeller**
PR for the purposes of reviewing documentation changes for planned modifications to mapcall agent.

Do not merge....

### CODE REVIEW

File: agent/mapcall/src/main/java/randoop/instrument/Premain.java [hunk 1]
```diff
@@ -44,8 +44,10 @@
    */
   public static void premain(String agentArgs, Instrumentation inst) throws IOException {
 
-    System.out.format(
-        "In premain, agentargs ='%s', " + "Instrumentation = '%s'%n", agentArgs, inst);
+    if (verbose) {
```
Comment (mernst) [inline agent/mapcall/src/main/java/randoop/instrument/Premain.java]: This seems unrelated to the rest of the changes.  How about making this change on master instead?  Then it won't show up in the diffs.

File: build.gradle [hunk 2]
```diff
@@ -256,13 +257,7 @@ test {
...
 }
 
-/*
- * Configuration of agentTest task.
- * Runs JUnit with exercised-class Java agent over all classes in agentTest
- * sourceSet.
- * Requires that exercised-class jar file copied into Randoop build/libs.
- */
-task agentTest(type: Test, dependsOn: 'copyJars') {
+task exercisedClassTest(type: Test, dependsOn: 'copyJars') {
```
Comment (mernst) [inline build.gradle]: Can you add brief documentation, as is given for other tasks?

File: docs/manual/index.html [hunk 3]
```diff
@@ -1063,66 +1063,145 @@ <h3 id="map_calls">Avoiding calls to specific methods</h3>
...
 <p>
-Alternately, you can provide a different implementation of a method that will be used
-when running under Randoop.  Randoop transforms the code under test,
-replacing each call to a undesired method by a call to a new implementation
-that you provide.
+Alternately, you can use the <code>mapcall</code> agent to provide a different
+implementation of a method that will be used when running under Randoop.
+The mapcall Java agent transforms the code under test, replacing each call to a
+undesired method by a call to a new implementation.
+Default alternate implementations are provided for <code>System.exit()</code>
```
Comment (mernst) [inline docs/manual/index.html]: Please avoid passive voice.  It isn't clear what a user needs to do to achieve this effect.

File: docs/manual/index.html [hunk 4]
```diff
@@ -1063,66 +1063,145 @@ <h3 id="map_calls">Avoiding calls to specific methods</h3>
 </p>
 
 <p>
-Alternately, you can provide a different implementation of a method that will be used
-when running under Randoop.  Randoop transforms the code under test,
-replacing each call to a undesired method by a call to a new implementation
-that you provide.
+Alternately, you can use the <code>mapcall</code> agent to provide a different
```
Comment (mernst) [inline docs/manual/index.html]: I would start out with the goal, rather than introducing jargon ("the mapcall agent") right at the very beginning of the description.

**Comment by bjkeller** [comment]
I'm closing this so we can switch to call-avoidance-docs-only branch
